### PR TITLE
Newssssssssss

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

**Comment by KillTheMule** [comment]
Ideas:

 
- mention https://github.com/neovim/neovim/pull/4489?

-  link to https://medium.com/@eric.burel/stop-using-open-source-5cb19baca44d#.dbg7pz8j1?

- include an invitation to chat/participate/... with links, so people feel welcome
- include something like a "More" section at the end where you point out the various resources available

- I barely have any idea if it's a good plugin, but [this](https://github.com/arakashic/chromatica.nvim) stuck out for me on reddit, maybe it's worth a mention for usage of buffer_add_highlight?

File: _posts/2016-07-01-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+Funding: Nikolai Pavlov (ZyX) assumes Thiago's role
+-------
+
+The enormously successful
+[salt campaign](https://salt.bountysource.com/teams/neovim) has funded
+Thiago's work, yielding libmpack, major refactors and improvements. For personal
+reasons Thiago will take a less active role in the project, and the 2-3 days of
+funded work will now be fulfilled by ZyX, the second-most [prolific](#note1)
+committer to Neovim _and Vim_ (~50,000 LOC contributed to Neovim, ~13,000 LOC
+contributed to Vim—including `if_pyth`, Vim's most mature FFI).
```
Comment (bfredl) [inline _posts/2016-07-01-newsletter.md]: maybe `if_pyth` is technically correct in some sense (the help file is called that at least), but doesn't `if_python` sound better?

File: _posts/2016-07-01-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+people based on reddit conversations)
+
+- `:terminal`
+- remote API
+- first-class XDG support
+- better defaults
+
+### New features
+
+- `buffer_add_highlight` API method. `:help api-highlights` position-based
```
Comment (bfredl) [inline _posts/2016-07-01-newsletter.md]: maybe not mention `buffer_add_highlight` by name as that name is soon deprecated. The `:help` reference is good enough and most (all?) existing plugins are not using that name directly but e.g. `buf.add_highlight` in python (which will not change).

File: _posts/2016-07-01-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+subtle problems that new text editors have yet to encounter, digest and decide.
+When a new text editor is built, the 80% case is solved neatly by today's more
+advanced frameworks and designs chosen with the benefit of hindsight. But the
+last 20% will take decades of tiny decisions, while yet more new text editors
+are written, and the old ones adapt...
+
+Neovim builds on the hard work put into Vim, and makes it viable for another 30
+years of hacking and extending. As always, consult
+[`:help vim-differences`](https://neovim.io/doc/user/vim_diff.html) for some
+relaxing beach--day reading about the user-facing changes and improvements in
```
Comment (brcolow) [inline _posts/2016-07-01-newsletter.md]: `beach--day` -> `beach-day` (just one hyphen)

File: _posts/2016-07-01-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+### New features
+
+- `buffer_add_highlight` API method. `:help api-highlights` position-based
+  highlights to buffers. This is similar to matchaddpos() but with some key
+  differences. The added highlights are associated with a buffer and adapts to
+  line insertions and deletions, similar to signs. It is also possible to manage
+  a set of highlights as a group and delete or replace all at once.
+
+  The intended use case are linter or semantic highlighter plugins that monitor
```
Comment (brcolow) [inline _posts/2016-07-01-newsletter.md]: maybe use case should be plural? `use cases`?

**Comment by justinmk** [comment]
pushed, some updates, still a ton of things to do.

File: _posts/2016-07-01-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+[debian](https://packages.debian.org/experimental/neovim).
+
+Progress
+--------
+
+What did we _really_ do?
+
+50,000 (TODO: exact number?) new lines of code have been written. That's 25% of the codebase we
+started with.
+1000 (TODO:?) _new_ tests (on top of Vim's existing test suite).
```
Comment (justinmk) [inline _posts/2016-07-01-newsletter.md]: would be helpful if someone could find out these actual numbers.

Comment (KillTheMule) [inline _posts/2016-07-01-newsletter.md]: Since the beginning of the project? Or since 0.1.4? Or since the last newsletter?

Comment (justinmk) [inline _posts/2016-07-01-newsletter.md]: Since the beginning. Note it should exclude Vim patch commits.

Comment (KillTheMule) [inline _posts/2016-07-01-newsletter.md]: Good, call, I"ll think about it :)

**Comment by KillTheMule** [comment]
```bash
rm -rf test/functional/legacy/
make functionaltest
1161 successes / 0 failures / 0 errors / 7 pending
make unittest
418 successes / 0 failures / 0 errors / 0 pending
```
That tells me we have added 1168 functional and 418 unit tests.

```bash
find test/functional/legacy -type f -name '*_spec.lua'|wc -l
122
```
We have converted 122 of the oldstyle test files to our new infrastructure. 15 of lucc's PR's are still open. After that, conversion will be done

```bash
git log --pretty=tformat: --numstat 72cf89bce8e4230dbc161dc5606f48ef9884ba70..HEAD| gawk '{ add += $1; subs += $2; loc += $1 - $2 } END { printf "added lines: %s removed lines: %s total lines: %s\n", add, subs, loc }' 
added lines: 1582078 removed lines: 1140037 total lines: 442041

git log --grep="vim-patch" --pretty=tformat: --numstat | gawk '{ add += $1; subs += $2; loc += $1 - $2 } END { printf "added lines: %s removed lines: %s total lines: %s\n", add, subs, loc }' 
added lines: 383760 removed lines: 20959 total lines: 362801

git log --pretty=tformat: --numstat -1 fb0ebb2a3a220a2e744efabed82beb08d88e158d| gawk '{ add += $1; subs += $2; loc += $1 - $2 } END { printf "added lines: %s removed lines: %s total lines: %s\n", add, subs, loc }' 
added lines: 46 removed lines: 42 total lines: 4
```

First command gives the total for [current master](https://github.com/neovim/neovim/commit/4d4c92535ab5249b8597e32f9f56d69c4e0c2e55) (the hash is the initial commit), next those those have `vim-patch` in their title. The last is the single commit I found that has `runtime` in it's title and really copies over stuff from vim (the other one's are just the merge commits for other `vim-patch` commits).

Did I forget something? I used science to get our current total:

```txt
added lines: 1198272  removed lines:  1119036 total lines: 79236
```

**Comment by fwalch** [comment]
Just some random notes:

Adding screenshots/gifs like https://github.com/neovim/neovim/pull/4432#issuecomment-240901890 could really help to illustrate our features (terminal emulator, GUIs, livesub, ..).

Also maybe include stuff like https://github.com/metakirby5/codi.vim/issues/5#issuecomment-240642210 (although that plugin doesn't seem to be fully working yet for everyone).

**Comment by romgrk** [comment]
If anyone has time and knowledge to convert a .webm to a better quality .gif that the one in the comment in #4432, here's a link to the original file: http://www.filedropper.com/cast

**Comment by justinmk** [comment]
@romgrk youtube?

**Comment by romgrk** [comment]
Damn right. Feel stupid now. Here you go. https://www.youtube.com/watch?v=TI5azVeDUDo

File: _posts/2016-07-01-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+- Users love [deoplete](https://github.com/Shougo/deoplete.nvim), the first and
+  (so far) only **non-blocking auto-completion** plugin for Vim.
+- [NyaoVim](https://github.com/rhysd/NyaoVim) by @rhysd is innovating and
+  exploring new concepts. But my favorite part is its thoughtful,
+  convention-based, modular design. It offers the
+  [`<neovim-editor>`](https://github.com/rhysd/neovim-component)
+  **web component** which can be used in your electron- or nw.js-based project,
+  including VS Code (someone get to work on this!) and of course Atom. Stuff
+  like this is exactly what we had hoped for 2 years ago when Neovim started...
+- @qvacua recently turned his sights to `nvim` and has made rapid progress a new
```
Comment (brcolow) [inline _posts/2016-07-01-newsletter.md]: `rapid progress a new` => `rapid progress on a new`

File: _posts/2016-07-01-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+  `nvim`-based backend for his well-known
+  **[VimR](https://github.com/qvacua/vimr)** project, a polished GUI frontend for macOS.
+- [neovim-qt](https://github.com/equalsraf/neovim-qt) continues to get better
+  and better: ts **performance matches gVim**, with **less flicker** when
```
Comment (brcolow) [inline _posts/2016-07-01-newsletter.md]: `ts` => `its'`

File: _posts/2016-07-01-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+  as `--servername`, `--remote`, etc. `nvr` is perfect for **communicating with
+  a parent `nvim` instance** from a `:terminal` buffer.
+- Intero [users are excited](https://twitter.com/_simonyang/status/753365931896692736) about
+  [intero.nvim](https://github.com/myfreeweb/intero.nvim) and
+  [intero-neovim](https://github.com/parsonsmatt/intero-neovim)
+- @carlosdcastillo continues to update the Neovim-based
+  [**Atom** nvim-mode](https://github.com/carlosdcastillo/vim-mode).
+
+There are clients for go, julia, perl, Java, R, Elixir, and Clojure. You really
+should go the [related projects](https://github.com/neovim/neovim/wiki/Related-projects)
```
Comment (blueyed) [inline _posts/2016-07-01-newsletter.md]: s/go/go to/

File: _posts/2016-07-01-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+
+### Funding: Nikolai Pavlov (ZyX) assumes Thiago's role
+
+The enormously successful
+[salt campaign](https://salt.bountysource.com/teams/neovim) has funded
+Thiago's work, yielding libmpack, major refactors and improvements. For personal
+reasons Thiago will take a less active role in the project, and the 2-3 days of
+funded work will now be fulfilled by ZyX, the second-most [prolific](#note1)
+committer to Neovim _and Vim_ (~50,000 LOC contributed to Neovim, ~13,000 LOC
+contributed to Vim—including `if_python`, Vim's most mature FFI).
```
Comment (blueyed) [inline _posts/2016-07-01-newsletter.md]: Expand FFI here maybe ("foreign function interface").

File: _posts/2016-07-01-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+a ton of time was put into the Neovim build and CI system. As many devops/SREs
+know, this is a major project by itself. An important part of a CI system is to
+"lock in" the gains by insisting on green builds and writing meaningful and
+robust tests.
+
+Now the years of work are really, really start to pay off. Instead of being
+petrified by far-reaching changes, we can welcome feature work large _and
+small_. In a stable but fragile C codebase, the tendency is to ignore on "small"
+features because they may not be popular and the maintainers don't have the
+incentive to tend to them.
```
Comment (blueyed) [inline _posts/2016-07-01-newsletter.md]: "tend to": "tendency" used before already.  Rephrase?

File: _posts/2016-07-01-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+      ~                        |
+      define  jump  list  >    |
+      :sign define^             |
+    ]])
+  end)
+end)
+```
+
+Having built a sophisticated build/CI system, the next "inflection point" of
+safe, rapid enhancement is to make the core extensible with lua. That is why
```
Comment (blueyed) [inline _posts/2016-07-01-newsletter.md]: s/lua/Lua/

File: _posts/2016-07-01-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+
+Each Neovim PR now builds against 12(!) different systems. Thanks to
+@jszakmeister, @fwalch and @ZyX-I for maturing the CI pipeline.
+Reliable and meaningful CI/automation is central to our
+"Move Fast and Don't Break Things™" methodology.
+
+We did break some things though. Many of those regressions have been fixed
+thanks to @oni-link's insight, and many others. Two years ago when we forked
+Vim, there was an unstable period; that gap has become smaller and smaller. Each
+of regression fix is covered by integration tests.
```
Comment (blueyed) [inline _posts/2016-07-01-newsletter.md]: s/of//

Comment (jamessan) [inline _posts/2016-07-01-newsletter.md]: `s/of //`

File: _posts/2016-07-01-newsletter.md [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+  put `%=` in your `'statusline'` setting the separate sections equally.
+
+- The `TextYankPost` event allows scripts to reliably and easily hook into any
+  yank ("text copy") event. This makes it trivial to implement a
+  [yank ring](https://github.com/bfredl/nvim-miniyank)—or even send yanks to an
+  external service.
+
+### Around the corner
+
+- ZyX lua work
```
Comment (blueyed) [inline _posts/2016-07-01-newsletter.md]: ZyX's Lua work

**Comment by fwalch** [comment]
Just a suggestion for reviewers: what about sending them as PRs to @justinmk's fork (https://github.com/justinmk/neovim.github.io/tree/newssssssssss) instead? Would save one read/type cycle :)

**Comment by justinmk** [comment]
Gah, the the one time I use `:Gpush`........

**Pull request merged**